urllib3/fields.py

```diff
@@ -36,11 +36,11 @@ def format_header_param(name, value):
         result = '%s="%s"' % (name, value)
         try:
             result.encode('ascii')
-        except UnicodeEncodeError:
+        except (UnicodeEncodeError, UnicodeDecodeError):
             pass
         else:
             return result
-    if not six.PY3:  # Python 2:
+    if not six.PY3 and isinstance(value, unicode):  # Python 2:
```

Comment:
Python 3 `str` type is the same as Python 2 `unicode` type. That's also why Flake8 is complaining BTW: there is no `unicode` type on Python 3.